src/Swift.Bindings/src/TypeDatabase/TypeDatabase.cs

```diff
@@ -1,39 +1,73 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+using System.Collections.Concurrent;
+using System.Diagnostics.CodeAnalysis;
 using System.Xml;
 
 namespace BindingsGeneration
 {
+
     /// <summary>
     /// Manages a mapping database between Swift types and C# types.
     /// </summary>
-    public unsafe class TypeDatabase
+    public class TypeDatabase : ITypeDatabase
```

Comment:
ModuleDatabase and TypeDatabase naming might be confusing. My first insight was that a module contain types, but type database contains modules.

Reply:
Sure, I am open to suggestions :)